src/components/fields/ObjectField.js

```diff
@@ -1,13 +1,32 @@
+import AddButton from "../AddButton";
 import React, { Component } from "react";
 import PropTypes from "prop-types";
 
 import {
   orderProperties,
   retrieveSchema,
   getDefaultRegistry,
+  getUiOptions,
 } from "../../utils";
 
 function DefaultObjectFieldTemplate(props) {
+  const canExpand = function canExpand() {
+    const { formData, schema, uiSchema } = props;
+    if (!schema.additionalProperties) {
+      return false;
+    }
+    const { expandable } = getUiOptions(uiSchema);
+    if (expandable !== false) {
+      // if ui:options.expandable was not explicitly set to false, we can add
+      // another property if we have not exceeded maxProperties yet
+      if (schema.maxProperties !== undefined) {
+        return Object.keys(formData).length < schema.maxProperties;
+      }
+      return true;
+    }
+    return expandable;
```

Comment:
You could refactor this as 

```
if (expandable === false) {
  return false;
}
if (schema.maxProperties !== undefined) {
  return Object.keys(formData).length < schema.maxProperties;
}
return true;
```

This would save a level of indentation. But that's just a personal preference.